backend/Application/AppGenerator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Application;
+
+use Internships\FileSystem\FileManager;
+use Internships\FileSystem\Path;
+use Internships\Services\CompanyDataFactory;
+use Internships\Services\CsvReader;
+use Internships\Services\DataFactory;
+use Internships\Services\FacultyDataFactory;
+
+class AppGenerator
+{
+    protected array $subFactories;
+
+    public function __construct(
+        protected FileManager $fileManager,
+        protected CsvReader $csvReader,
+        protected FacultyDataFactory $facultyFactory,
+        CompanyDataFactory $companyDataFactory,
+    ) {
+        $this->subFactories = [
+            $companyDataFactory,
+        ];
+    }
+
+    public function getDataFromCsv(DataFactory $factory): array
+    {
+        $facultyCsvData = $this->csvReader->getCSVData(
+            $factory->getSourceRelativePath(),
+            $factory->getSourceFileName(),
+            $factory->getFields()
+        );
+        return $factory->buildFromData($facultyCsvData);
+    }
+
+    public function saveDataToJson(DataFactory $factory, array $data): void
+    {
+        $this->fileManager->create(
+            $factory->getDestinationRelativePath(),
+            $factory->getDestinationFileName(),
+            json_encode($data, $this->fileManager->getDefaultJsonFlags())
+        );
+    }
+
+    public function generateStaticData(): void
+    {
+        $faculties = $this->getDataFromCsv($this->facultyFactory);
+        $this->saveDataToJson($this->facultyFactory, $faculties);
+        foreach ($faculties as $faculty) {
+            foreach ($this->subFactories as $subFactory) {
+                echo "Processing " . $faculty->getDirectory() . "." . PHP_EOL;
```

Comment:
Maybe create a some class/helper which will write the output?
Something like `OutputWriter`
OutputWriter::newLine(`test);

This static method will add `echo` and `PHP_EOL` by default, so you will avoid adding this everytime.

Reply:
I think I'll leave "echo" out from the function. This way it can be reused when writing data into file, which I'll use later.

Reply:
I prefer create separate method for write to the console, and write to the file.

Reply:
Alright. Console output will be separate.

I think writing newLine to file should be handled through FileManager, because it's already configured to safely handle those operations and can be extended for appending data. I can simply use method from OutputWriter in there.

This way we don't have to introduce unnecessary dependency.